src/components/Layout/Header/Header.tsx

```diff
@@ -122,18 +125,26 @@ export const Header = memo(() => {
       isSnapInstalled === false &&
       previousIsCorrectVersion === true
     ) {
-      // they uninstalled the snap
-      toast({
-        status: 'success',
-        title: translate('walletProvider.metaMaskSnap.snapUninstalledToast'),
-        position: 'bottom',
-      })
+      if (connectedRdns === METAMASK_RDNS) {
+        // they uninstalled the snap
+        toast({
+          status: 'success',
+          title: translate('walletProvider.metaMaskSnap.snapUninstalledToast'),
+          position: 'bottom',
+        })
+      }
       const walletId = currentWalletId
       if (!walletId) return
       appDispatch(portfolio.actions.clearWalletMetadata(walletId))
-      return snapModal.open({ isRemoved: true })
+      if (connectedRdns === METAMASK_RDNS) {
+        return snapModal.open({ isRemoved: true })
+      }
     }
-    if (previousSnapInstall === false && isSnapInstalled === true) {
+    if (
+      previousSnapInstall === false &&
+      isSnapInstalled === true &&
+      connectedRdns === METAMASK_RDNS
+    ) {
```

Comment:
blocking: there is a bug with this condition, if I have the snap previously installed, it pass through this condition while it shouldn't, I think `previousSnapInstall` is false while it should be true

https://jam.dev/c/f3365f85-9200-4387-b7cb-3ac96e41fa28